src/uint/boxed/mul.rs

```diff
@@ -24,7 +38,17 @@ impl BoxedUint {
 
     /// Multiply `self` by itself.
     pub fn square(&self) -> Self {
-        let mut limbs = vec![Limb::ZERO; self.nlimbs() * 2];
+        let size = self.nlimbs() * 2;
+
+        if self.nlimbs() >= 64 {
```

Comment:
Likewise here, this could probably also use a constant (or short expression like `KARATSUBA_MIN_LIMBS * 2` or thereabouts)